main.js

```diff
@@ -29,3 +29,67 @@ menuOptions.addEventListener('click', () => {
   menuOptions.classList.remove('open');
   openMenu.classList.add('fa-bars');
 });
+
+const objectContainer = [];
+function CardConstruct(
+  title,
+  timeline,
+  picture,
+  description,
+  technologies,
+  liveURL,
+  sourceURL,
+) {
+  this.title = title;
+  this.timeline = timeline;
+  this.picture = picture;
+  this.description = description;
+  this.technologies = technologies;
+  this.liveURL = liveURL;
+  this.sourceURL = sourceURL;
+}
+
+const projects = document.querySelectorAll('.project');
+
+for (let project = 0; project < projects.length; project += 1) {
+  const title = projects[project].querySelector('.project-title').innerText;
+  const timeline = projects[project].querySelector('.project-data').innerHTML;
+  const picture = projects[project].querySelector('img').src;
+  const description = projects[project].querySelector('.project-description').innerText;
+  const technologies = projects[project].querySelector('.buttons-flex').innerHTML;
+  const liveURL = 'https://oscarfmdev.github.io/PortfolioProject/';
+  const sourceURL = 'https://github.com/OscarFMdev/PortfolioProject';
+  objectContainer.push(
+    new CardConstruct(
+      title,
+      timeline,
+      picture,
+      description,
+      technologies,
+      liveURL,
+      sourceURL,
+    ),
+  );
+}
+
+const modalButton = document.querySelectorAll('.project-button');
+const modal = document.querySelector('.modal-container');
+const closeModal = document.querySelector('.close-modal');
+for (let index = 0; index < modalButton.length; index += 1) {
+  modalButton[index].setAttribute('data-index', index);
+  modalButton[index].addEventListener('click', (e) => {
+    const pos = e.target.getAttribute('data-index');
+    modal.querySelector('.modal-title').innerHTML = objectContainer[pos].title;
+    modal.querySelector('.project-data').innerHTML = objectContainer[pos].timeline;
+    modal.querySelector('.img-container img').src = objectContainer[pos].picture;
+    modal.querySelector('.desc').innerHTML = objectContainer[pos].description;
+    modal.querySelector('.buttons-flex').innerHTML = objectContainer[pos].technologies;
+    modal.querySelector('.live').href = objectContainer[pos].liveURL;
+    modal.querySelector('.source').href = objectContainer[pos].sourceURL;
+    modal.style.display = 'block';
+  });
+}
+
+closeModal.addEventListener('click', () => {
+  modal.style.display = 'none';
+});
```

Comment:
- [ ] You have implemented your project well but you are not following the requirements :fearful: Kindly, make an array of objects and each object will contain the information about your project card.
- [ ] You have to use the same array of objects for your popup modal window :+1:

Reply:
Dear @HAMMAS-SALEEM, it would be better that instead of commenting on the entire document, read the code.
here you can see on line 62 that we have an array and we have pushed the cards object inside the array.